### DISCUSSION TITLE
Support for multiple versions

### DISCUSSION THREAD

**Discussion by kvanzuijlen** [renovatebot/renovate#26489] [Request Help]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

37.118.0

### If you're self-hosting Renovate, select which platform you are using.

github.com

### What is your question?

I was wondering if it's possible to keep the last n minor/major versions up to date. For example, if I have the following list:
```yaml
          - "kindest/node:v1.29.0"
          - "kindest/node:v1.28.0"
          - "kindest/node:v1.27.2"
          - "kindest/node:v1.26.4"
```
and v1.29.1 comes out, I'd like the resulting PR to be
```yaml
          - "kindest/node:v1.29.1"
          - "kindest/node:v1.28.0"
          - "kindest/node:v1.27.2"
          - "kindest/node:v1.26.4"
```
After that PR is merged and v1.30.0 comes out I'd like the resulting PR to be:
```yaml
          - "kindest/node:v1.30.0"
          - "kindest/node:v1.29.1"
          - "kindest/node:v1.28.0"
          - "kindest/node:v1.27.2"
```

Also, if at the same time of v1.30.0 coming out, v1.29.2 came out, the resulting PR should be:
```yaml
          - "kindest/node:v1.30.0"
          - "kindest/node:v1.29.2"
          - "kindest/node:v1.28.0"
          - "kindest/node:v1.27.2"
```

I know this is kind of complicated to do and probably currently not possible, but maybe this makes for a good feature request?

Examples for where this could come in useful:
- [jenkinsci/helm-charts - .github/workflows/lint-test.yaml](https://github.com/jenkinsci/helm-charts/blob/255e739d13a9f30cf84265906140eb4f6bbd9798/.github/workflows/lint-test.yaml#L14C1-L17C35)
- [renovatebot/helm-charts - .github/workflows/ci.yaml](https://github.com/renovatebot/helm-charts/blob/19273aa394c0eee43be55d734d4abad2acb3f577/.github/workflows/ci.yaml#L60C1-L64C20)

**Comment by viceice** [comment]
the patch versions can already be update with regex manager, but the other thing is not possible.

**Reply by kvanzuijlen** [reply]
Yes. I'm familiar with the regex manager, but this request is mainly about keeping the most recent `n` major/minor/patch versions. Is this something renovate could at one point support or is this out-of-scope?

**Reply by rarkins** [reply]
I can't think of any elegant way to support "last n" so there's no plan to support this one day

**Comment by dklimpel** [comment]
Not the same but a little bit related to:
- #25248
